### PR TITLE
reverts upgrade to tokio-native-tls

### LINKED ISSUE #71: recent upgrade to tokio-native-tls breaks crates that depend on hyper-tls
i see this crate was released an hour ago - seems to have broken my crates relying on `reqwest`.

see https://github.com/seanmonstar/reqwest/issues/971 for report on reqwest crate

more specifically - it looks like `std::convert::From<(connect::HttpConnector, tokio_tls::TlsConnector)>` is no longer implemented (i think [this commit](https://github.com/hyperium/hyper-tls/commit/7c9adcdebba980ed3d3a6190480fafe37628264e)) which is causing this issue

### PR DESCRIPTION
**Pull request by EverlastingBugstopper**
fixes #71 - breaking change introduced in 0.4.2

### CODE REVIEW

File: src/stream.rs [hunk 1]
```diff
@@ -6,7 +6,7 @@ use std::task::{Context, Poll};
 use bytes::{Buf, BufMut};
 use hyper::client::connect::{Connected, Connection};
 use tokio::io::{AsyncRead, AsyncWrite};
-pub use tokio_native_tls::TlsStream;
```
Comment (EverlastingBugstopper) [inline src/stream.rs:9]: i believe this here is the line that broke dependencies

**Comment by EverlastingBugstopper** [comment]
cc @LucioFranco and @seanmonstar

**Comment by LucioFranco** [comment]
Ugh sigh sneaky `From` impl, @seanmonstar I wonder if we can just yank that version and publish a 0.3 with master's commits?

**Comment by seanmonstar** [comment]
Thanks for filing this! I wanted to try to keep the IPv6 patch and this was removing it too, so I merged #73.

**Pull request closed without merging**